src/Connector.sol

```diff
@@ -167,25 +182,27 @@ contract CentrifugeConnector {
         emit TrancheDeployed(poolId, trancheId, token);
     }
 
-    function updateTokenPrice(uint64 poolId, bytes16 trancheId, uint128 price) public onlyRouter {
+    function updateTokenPrice(uint64 poolId, bytes16 trancheId, uint128 price) public onlyGateway {
         Tranche storage tranche = tranches[poolId][trancheId];
         require(tranche.lastPriceUpdate > 0, "CentrifugeConnector/invalid-pool-or-tranche");
         tranche.latestPrice = price;
         tranche.lastPriceUpdate = block.timestamp;
     }
 
-    function updateMember(uint64 poolId, bytes16 trancheId, address user, uint64 validUntil) public onlyRouter {
+    function updateMember(uint64 poolId, bytes16 trancheId, address user, uint64 validUntil) public onlyGateway {
         Tranche storage tranche = tranches[poolId][trancheId];
         require(tranche.lastPriceUpdate > 0, "CentrifugeConnector/invalid-pool-or-tranche");
         RestrictedTokenLike token = RestrictedTokenLike(tranche.token);
         MemberlistLike memberlist = MemberlistLike(token.memberlist());
         memberlist.updateMember(user, validUntil);
     }
 
-    function handleTransfer(uint64 poolId, bytes16 trancheId, address destinationAddress, uint128 amount)
+    function handleTransfer(uint64 poolId, bytes16 trancheId, address destinationAddress, uint256 chainId, uint128 amount)
```

Comment:
Seems to me to make more sense to have chain before address. And then I would name `chainId` to `destinationChainId`